app/controllers/contacts_controller.rb

```diff
@@ -18,28 +18,49 @@ def update
   end
 
   def create
-    if Role.exists?(name: params[:name].downcase)
-      flash[:notice] = "Team '#{params[:name]}' already exists!"
+    role = Role.where('name ILIKE ?', params[:name]).first
+    if role.present? && !role.archived # if archived, un-archive
+      flash[:alert] = "Team '#{role.name}' already exists!"
     else
-      Role.create(name: params[:name].downcase)
-      flash[:success] = "Successfully added new team #{params[:name].capitalize}"
+      update_or_create_role(role)
     end
     redirect_to action: 'index'
   end
 
   def destroy
-    to_destroy = Role.where(name: params[:name].downcase).first
-    if to_destroy
-      to_destroy.destroy
-      flash[:success] = "Successfully removed #{params[:name].capitalize} team."
-    else
-      flash[:alert] = "Could not find team #{params[:name].capitalize}!"
-    end
+    flash[:danger] = 'Could not delete teams at this time.' unless destroy_roles
     redirect_to action: 'index'
   end
 
   private
 
+  def destroy_roles
```

Comment:
![screen shot 2017-12-05 at 1 26 37 pm](https://user-images.githubusercontent.com/26726721/33626373-041e104a-d9c0-11e7-861e-895356ff7d48.png)
If I click "select teams to delete" and then click cancel without deleting or selecting, I get this error

Reply:
If I do select a team to delete and then click cancel, it still deletes the team

Reply:
@rbrezynski Can you try it out again? I've fixed the errors.